py4ai/core/data/layer/v1/pandas/repository.py

```diff
@@ -0,0 +1,222 @@
+"""Module with abstraction for accessing to data persistent in pickles, mimicking a ficticious database."""
+
+from abc import abstractmethod, ABC
+from functools import cached_property
+from typing import Optional, Tuple, TypeVar, Sequence
+
+import pandas as pd
+from pandas.errors import EmptyDataError
+
+from py4ai.core.data.layer.v1.common.criteria import SearchCriteria
+from py4ai.core.data.layer.v1.common.repository import (
+    AbstractRepository,
+    KE,
+    E,
+    Q,
+    QueryOptions,
+    Paged,
+    SortingDirection,
+)
+from py4ai.core.data.layer.v0.pandas.archivers import PandasArchiver
+from py4ai.core.data.layer.v1.common.serialiazer import DataSerializer
+from py4ai.core.data.layer.v1.pandas.criteria import PandasSearchCriteria, PandasFilter
+from py4ai.core.typing import PathLike
+
+T = TypeVar("T", bound=PandasArchiver)
+KD = TypeVar("KD", str, int, Tuple)
+
+
+class PandasRepository(AbstractRepository[KE, KD, E, pd.Series, PandasFilter], ABC):
+    """Archiver based on persistent layers based on tabular files, represented in memory by a pandas DataFrame."""
```

Comment:
I don't get this explanation... What do you mean exactly with "Archiver"?

Reply:
Changed to Repository